dsc/pwsh.profile.dsc.resource.json

```diff
@@ -0,0 +1,101 @@
+{
+    "$schema": "https://aka.ms/dsc/schemas/v3/bundled/resource/manifest.json",
+    "description": "Manage PowerShell profiles.",
+    "tags": [
+        "linux",
+        "windows",
+        "macos",
+        "powershell"
+    ],
+    "type": "Microsoft.PowerShell/Profile",
+    "version": "0.1.0",
+    "get": {
+        "executable": "pwsh",
+        "args": [
+            "-NoLogo",
+            "-NonInteractive",
+            "-NoProfile",
+            "-ExecutionPolicy",
+            "Bypass",
+            "-Command",
+            "$Input | ./pwsh.profile.resource.ps1 -operation 'get'"
+        ],
+        "input": "stdin"
+    },
+    "set": {
+        "executable": "pwsh",
+        "args": [
+            "-NoLogo",
+            "-NonInteractive",
+            "-NoProfile",
+            "-ExecutionPolicy",
+            "Bypass",
+            "-Command",
+            "$Input | ./pwsh.profile.resource.ps1 -operation 'set'"
+        ],
+        "input": "stdin"
+    },
+    "export": {
+        "executable": "pwsh",
+        "args": [
+            "-NoLogo",
+            "-NonInteractive",
+            "-NoProfile",
+            "-ExecutionPolicy",
+            "Bypass",
+            "-Command",
+            "$Input | ./pwsh.profile.resource.ps1 -operation 'export'"
+        ],
+        "input": "stdin"
+    },
+    "schema": {
+        "embedded": {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "title": "Profile",
+            "description": "A resource for managing PowerShell profiles.",
+            "type": "object",
+            "unevaluatedProperties": false,
+            "required": [
+                "profileType"
+            ],
+            "properties": {
+                "profileType": {
+                    "$ref": "#/$defs/ProfileType"
+                },
+                "content": {
+                    "title": "Content",
+                    "description": "The content of the profile.",
+                    "type": "string"
+                },
+                "_exist": {
+                    "$ref": "https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/v3/resource/properties/exist.json"
+                }
+            },
+            "$defs": {
+                "ProfileType": {
+                    "type": "string",
+                    "title": "Profile Type",
+                    "description": "The type of the profile. Can be 'AllUsersCurrentHost', 'AllUsersAllHosts', 'CurrentUserAllHosts', or 'CurrentUserCurrentHost'.",
+                    "enum": [
+                        "AllUsersCurrentHost",
+                        "AllUsersAllHosts",
+                        "CurrentUserAllHosts",
+                        "CurrentUserCurrentHost"
+                    ]
```

Comment:
Non-blocking nit: Would prefer to have this definition in-lined, since it's not reused (or a reference to an external schema that we're bundling).